livekit-plugins/livekit-plugins-google/livekit/plugins/google/tts.py

```diff
@@ -0,0 +1,123 @@
+    def __init__(
+        self,
+        config: Optional[TTSOptions] = None,
+        *,
+        language: LgType = "en-US",
+        gender: GenderType = "neutral",
+        voice_name: str = "",  # Not required
+        audio_encoding: AudioEncodingType = "wav",
+        sample_rate: int = 24000,
+        speaking_rate: float = 1.0,
+        credentials_info: Optional[dict] = None,
+        credentials_file: Optional[str] = None,
+    ) -> None:
+        super().__init__(streaming_supported=False)
+
+        if credentials_info:
+            self._client = (
+                texttospeech.TextToSpeechAsyncClient.from_service_account_info(
+                    credentials_info
+                )
+            )
+        elif credentials_file:
+            self._client = (
+                texttospeech.TextToSpeechAsyncClient.from_service_account_file(
+                    credentials_file
+                )
+            )
+        else:
+            self._client = texttospeech.TextToSpeechAsyncClient()
+
+        if not config:
+            _gender = SsmlVoiceGender.NEUTRAL
+            if gender == "male":
+                _gender = SsmlVoiceGender.MALE
+            elif gender == "female":
+                _gender = SsmlVoiceGender.FEMALE
+            voice = texttospeech.VoiceSelectionParams(
+                name=voice_name,
+                language_code=language,
+                ssml_gender=_gender,
+            )
+            # Default audio to LINEAR16 (the encoding used in WAV files).
+            _audio_encoding = texttospeech.AudioEncoding.LINEAR16
+            if audio_encoding == "mp3":
+                _audio_encoding = texttospeech.AudioEncoding.MP3
+            elif audio_encoding == "opus":
+                _audio_encoding = texttospeech.AudioEncoding.OGG_OPUS
+            elif audio_encoding == "mulaw":
+                _audio_encoding = texttospeech.AudioEncoding.MULAW
+            elif audio_encoding == "alaw":
+                _audio_encoding = texttospeech.AudioEncoding.ALAW
+            config = TTSOptions(
+                voice=voice,
+                audio_config=texttospeech.AudioConfig(
+                    audio_encoding=_audio_encoding,
+                    sample_rate_hertz=sample_rate,
+                    speaking_rate=speaking_rate,
+                ),
+            )
+        self._config = config
+        self._creds = self._client.transport._credentials  # TODO: is this needed?
+
+    async def synthesize(
+        self,
+        *,
+        text: str,
+    ) -> tts.SynthesizedAudio:
+        # Perform the text-to-speech request on the text input with the selected
+        # voice parameters and audio file type
+        response: SynthesizeSpeechResponse = await self._client.synthesize_speech(
+            input=texttospeech.SynthesisInput(text=text),
+            voice=self._config.voice,
+            audio_config=self._config.audio_config,
+        )
+
+        data = response.audio_content
```

Comment:
This only works for linear16 right? Other formats would need to be decoded

Reply:
If the data needs to be in linear16, yes we would probably need a conversion from other formats.  Do you have any suggestions on libraries to use for this, there is [opuspy](https://github.com/elevenlabs/opuspy) that wraps the opus cpp library for example.

Perhaps in the short term we could restrict to only linear16 format?

Reply:
@keepingitneil i've update the google TTS to add support for mp3 codec, and let the others commented out until we want to add support for them in the agents codecs library.  I've also fixed tests to return async generator.

Reply:
This looks good! Thanks for doing this! I'll merge it, probably won't make it into a release for a week or so (we need to fix some broken tests on our end)